### ISSUE TITLE
How to add transform in validation code.

### ISSUE THREAD

**Issue by PapaMadeleine2022** [OML-Team/open-metric-learning#338]
Hello,  After training my model and getting weights, I want to do validation.  Here is my code:

```
import torch
from tqdm import tqdm

from oml.datasets.base import DatasetQueryGallery
from oml.metrics.embeddings import EmbeddingMetrics
from oml.registry.transforms import get_transforms_for_pretrained
from oml.transforms.images.albumentations.transforms import (
    get_augs_albu,
    get_normalisation_albu,
    get_normalisation_resize_albu,
    get_normalisation_resize_albu_clip,
)


df_val = ...

model = CustomModel().eval()
transforms= get_normalisation_resize_albu(im_size=224)
val_dataset = DatasetQueryGallery(df_val, transform = transforms)

val_loader = torch.utils.data.DataLoader(val_dataset, batch_size=4)
calculator = EmbeddingMetrics()
calculator.setup(num_samples=len(val_dataset))

with torch.no_grad():
    for batch in tqdm(val_loader):
        batch["embeddings"] = model(batch["input_tensors"])
        calculator.update_data(batch)

metrics = calculator.compute_metrics()
```
But I wonder if the function of `transform` I add is same with the function of `norm_resize_albu` in https://github.com/OML-Team/open-metric-learning/blob/main/examples/extractor_cub/train_cub.yaml#L22 ?

And the result should not has diff?

**Comment by AlekseySh** [comment]
Hey, 

First, you create two `val_dataset`. Please, double-check it.

Second, (probably you know it, but just for clarification and for other people who may read it) we should distinguish transforms for train and for validation. Train transforms are usually augmentations that change images randomly, but validation transforms are deterministic and usually contain only normalization and resizing.  `get_transforms_for_pretrained` returns validation transforms.

Third, the transforms that you get in your script are: https://github.com/OML-Team/open-metric-learning/blob/main/oml/registry/transforms.py#L52, the ones from link are: https://github.com/OML-Team/open-metric-learning/blob/main/oml/registry/transforms.py#L23. So, they are different.

**Comment by PapaMadeleine2022** [comment]
@AlekseySh Thanks for your reply. I have modified my question.

**Comment by PapaMadeleine2022** [comment]
> @PapaMadeleine2022 , offtopic: I would like to learn more about your way of using OML, your feedback is really useful for us. If you want to be in closer touch, please, message me in telegram: @Lexibender.

Generally, when we use metric learning on own train_dataset, we want to get a low-dim feature, so I would not entirely use  the original classical model like:
```
model = ViTExtractor("vits16_dino", arch="vits16", normalise_features=False).train()
```
I have to change the  he original classical  model and custom my model: 
```
class CustomModel(IExtractor):

    def __init__(self, pretrained):
        super().__init__()
        # self.resnet = resnet50(pretrained=pretrained)
        self.resnet = resnet50(weights=ResNet50_Weights.IMAGENET1K_V2)
        self.gem_p = 5.0
        self.resnet.avgpool = GEM(p=self.gem_p)
        self.resnet.fc =  nn.Linear(2048, 256)

        state_dict = torch.load(weights, map_location="cpu")
        state_dict = state_dict["state_dict"] if "state_dict" in state_dict.keys() else state_dict
        state_dict = remove_prefix_from_state_dict(state_dict, "layer4.")  # type: ignore
        self.model.load_state_dict(state_dict, strict=True)
        
    def forward(self, x):
        return self.resnet(x)

    @property
    def feat_dim(self):
        return self.resnet.fc.out_features
```
And use the `train.yaml` to train my train_dataset. After we get the weights, generally we want to extract features on our test_dataset. `How to use models from Zoo?` and `Validation`  are useful examples, but the `transform` of both need to pay attention to that we have to learn more about `oml` code. A newer would easily make mistakes and would not align the results with the model he just trained by `train.yaml` when using `Validation` example code.

**Comment by PapaMadeleine2022** [comment]
> Now the results have to be aligned.

Thanks. Got it!

**Comment by AlekseySh** [comment]
@PapaMadeleine2022 Thank you for providing the details! I will think about how to make work with transforms easier.
